src/main/java/bibliotecame/back/Review/ReviewController.java

```diff
@@ -58,15 +58,8 @@ public ReviewController(ReviewService reviewService, UserService userService, Bo
 
     private boolean userPreviouslyBookedThisOne(Integer bookId){
         List<LoanModel> loans = getLogged().getLoans();
-        List<CopyModel> copies = bookService.findBookById(bookId).getCopies();
-        List<String> copiesIds = new ArrayList<>();
-        for (int i = 0; i < copies.size() ; i++) {
-            copiesIds.add(copies.get(i).getId());
-        }
-        for (int i = 0; i < loans.size() ; i++) {
-            if(copiesIds.contains(loans.get(i).getCopy().getId())) return true;
-        }
-        return false;
+        List<String> copiesIds = bookService.findBookById(bookId).getCopies().stream().map(CopyModel::getId).collect(Collectors.toList());
+        return loans.stream().map(loanModel -> copiesIds.contains(loanModel.getCopy().getId())).reduce(true, ((aBoolean, aBoolean2) -> aBoolean && aBoolean2));
```

Comment:
This line is not working as expected. 
**What is checking:**
If ALL of the user the loans has a copy that belongs to the book given.

**What should be checking:**
If ANY of the user loans has a copy that belongs to the book given.

**How should be fixed**
Changing the `reduce`, the `initial value` must be false and the operation should be `OR (||)`. But please, check the correct working of the method, do some test with postman (user with more than 1 loan).